core/src/main/java/jenkins/plugins/DetachedPluginsManager.java

```diff
@@ -93,34 +101,32 @@ private DetachedPluginsManager() {
     }
 
     /**
-     * Get the list of all plugins that have ever been {@link DetachedPlugin detached} from Jenkins core.
+     * Get the list of all plugins that have ever been {@link DetachedPlugin detached} from Jenkins core, applicable to the current Java runtime.
      *
      * @return A {@link List} of {@link DetachedPlugin}s.
+     * @see JavaUtils#getCurrentJavaRuntimeVersionNumber()
      */
     @Restricted(NoExternalUse.class)
     public static @Nonnull
     List<DetachedPlugin> getDetachedPlugins() {
-        return DETACHED_LIST;
+        return DETACHED_LIST.stream()
```

Comment:
Can't this be done in the initialization of the DETACHED_LIST directly?  If possible, we don't need extra changes along the class changing DETACHED_LIST by getDetachedPlugins

Reply:
Yeah, I considered it already, not 100% sure why I chose this path. I'll have another look.